lib/filewatch/sincedb_value.rb

```diff
@@ -0,0 +1,82 @@
+require 'filewatch/boot_setup' unless defined?(FileWatch)
+
+module FileWatch
+  class SincedbValue
+    EXP_SECONDS = SDB_EXPIRES_DAYS * 24 * 3600
+    attr_reader :expires, :watched_file
+
+    def initialize(position, expires = nil, second_fp = nil, watched_file = nil)
+      @position = position # this is the value read from disk
+      @expires = expires
+      @second_fp = second_fp # from sincedb file read
+      @watched_file = watched_file
+      upd_expiry if @expires.nil? || @expires.zero?
+    end
+
+    def position
+      # either the value from disk or the current wf position
+      if @watched_file.nil?
+        @position
+      else
+        @watched_file.bytes_read
+      end
+    end
+
+    def upd_position(pos)
+      upd_expiry
+      if @watched_file.nil?
+        @position = pos
+      else
+        @watched_file.update_bytes_read(pos)
+      end
+    end
+
+    def inc_position(pos)
+      upd_expiry
+      if watched_file.nil?
+        @position += pos
+      else
+        @watched_file.increment_bytes_read(pos)
+      end
+    end
+
+    def upd_watched_file(wf)
+      upd_expiry
+      @watched_file = wf
+    end
+
+    def upd_expiry
+      @expires = new_expires
+    end
```

Comment:
refactoring to use last_seen_at and NOT to pre-calculate and store a future timestamp in the sincedb (after convergence discussions with beats team)